### PR TITLE
Update hhr.py to consider incomplete confidence

### PR DESCRIPTION
**Pull request by whatever60**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #...

### CODE REVIEW

**Comment by whatever60** [comment]
Current logic of parsing the confidence field in Bio/Align/hhr.py simply removes all internal spaces and adds spaces at the beginning and ending of the string. However, this results in error when a whole line of confidence field is empty. Besides, internal spaces might correspond to gap positions in query.

New code could handle these cases.

In my local unit test, test for hhr passed but few other irrelevant Biopython modules failed.

File: Bio/Align/hhr.py [hunk 1]
```diff
@@ -107,7 +107,10 @@ def create_alignment():
             )
             target.letter_annotations["ss_pred"] = fmt % target_ss_pred.replace("-", "")
             target.letter_annotations["ss_dssp"] = fmt % target_ss_dssp.replace("-", "")
-            target.letter_annotations["Confidence"] = fmt % confidence.replace(" ", "")
+            # target.letter_annotations["Confidence"] = fmt % confidence.replace(" ", "")
```
Comment (mdehoon) [inline Bio/Align/hhr.py:110]: please remove this line

**Comment by mdehoon** [comment]
@peterjc  Would you know why the ci/circleci and continuous-integration/appveyor/pr don't run?

**Comment by peterjc** [comment]
Curious. There was a GitHub outage recenltly - maybe that was why? Probably a safe bet to merge anyway...

**Comment by whatever60** [comment]
haha I experienced the outage exactly before I did the last push. I could do another dummy commit if that may help.

**Comment by peterjc** [comment]
Sure - how about you fix the alphbetical sorting of the contributors for the next release in NEWS.rst?

**Comment by whatever60** [comment]
oh sure. Didn't realize it is alphabetical order.

**Comment by peterjc** [comment]
Tests running now...

**Comment by mdehoon** [comment]
Thanks! That did the trick.

**Comment by peterjc** [comment]
Looks like a glitch in the merge accidentally undid commit 8345c7e430204bc4f47559aaaea176ffbd0833da causing #4338 to fail. Should be fixed in 2287f70c0c54b893900442abd861735ac01c35bd

**Pull request merged**